src/server/luaentity_sao.cpp

```diff
@@ -50,7 +50,14 @@ LuaEntitySAO::LuaEntitySAO(ServerEnvironment *env, v3f pos, const std::string &d
 		rotation.X = readF1000(is);
 		rotation.Z = readF1000(is);
 
-		// if (version2 < 2)
+		if (version2 < 2) {
+			m_guid = env->getGUIDGenerator().next();
```

Comment:
Q: will an object be saved asap if its static data changes due to now receiving a GUID?

Reply:
GUID is generated in the `LuaEntitySAO` constructor.

So when the entity is saved, it should be saved with a GUID.

Reply:
I answered my own question: since the guid is part of the static data the object will always be re-saved to the map
https://github.com/luanti-org/luanti/blob/fdc149f31606a2f9c2bf572a7ecc9176c7b19476/src/serverenvironment.cpp#L1715